scripts/cluster-install/openstack_connection.py

```diff
@@ -0,0 +1,127 @@
+import time
+import traceback
+import openstack
+
+
+def _get_client_config(config):
+    client_config = {
+        'auth_url': config['auth_url'],
+        'username': config['username'],
+        'password': config['password'],
+        'region_name': config['region_name'],
+        'project_name': config['tenant_name'],
+        'user_domain_name': 'Default',
+        'project_domain_id': 'Default'
+    }
+    return client_config
+
+
+def _get_resource_config(config, name):
+    resource_config = {
+        'name': name,
+        'image_id': '70de1e0f-2951-4eae-9a8f-05afd97cd036',  # centos image_id
+        'flavor_id': '3',  # m1.medium id
+        'networks': [{'uuid': config['network_id']}],
+        'key_name': config['key_name']
+    }
+    return resource_config
+
+
+def _create_sec_group_and_open_all_ports(connection):
+    sec_group = connection.network.create_security_group(
+        name='cluster-sec-group')
+    connection.network.create_security_group_rule(
+        security_group_id=sec_group.id,
+        direction='ingress',
+        remote_ip_prefix='0.0.0.0/0',
+        protocol='tcp',
```

Comment:
Given the intent of this, can you instead make it open 'all' or 'any' (or whatever it is), so that it also allows ICMP (ping)- or does that have to be separate?

Reply:
I created a new rule for the ICMP protocol.